tasks/less.js

```diff
@@ -51,6 +52,23 @@ module.exports = function(grunt) {
       var compiled = [];
       var i = 0;
 
+      if(typeof(options.combineSourceFiles) === "boolean" && options.combineSourceFiles){
+        var compound = "";
+        tempFileName = (function() {
+          var name = "x";
+          while(grunt.file.exists(name+".less")){
+            name += "x";
+          }
+          return name+".less";
+        }());
+
+        files.forEach(function(el) {
```

Comment:
This won't work if there are excludes in the files array (!myfile/path.less)  will probably change this a bit.